src/ir/local-graph.h

```diff
@@ -178,7 +178,11 @@ struct LocalGraph : public LocalGraphBase {
 struct LocalGraphFlower;
 
 struct LazyLocalGraph : public LocalGraphBase {
-  LazyLocalGraph(Function* func, Module* module = nullptr);
+  // We optionally receive an expression class to consider relevant for obstacle
+  // queries, see below. (Only obstacles of this class can be queried later.)
+  LazyLocalGraph(Function* func,
+                 Module* module = nullptr,
+                 std::optional<Expression::Id> obstacleClass = std::nullopt);
```

Comment:
Taking a single expression ID seems oddly restrictive for a general utility like this. Could we make this a general predicate function instead, perhaps?

Reply:
A predicate function is possible, but would be slower I think.

I think this API is not odd, though: it is reasonable to consider a particular expression. For example, in HeapStoreOptimizations we always want to move LocalSets to StructSets. Each optimization has a pattern it optimizes, and a particular expression type that we move to. We may want to generalize this more later, of course.